nilearn/_utils/fixes/__init__.py

```diff
@@ -10,11 +10,8 @@
     from sklearn.feature_selection import f_regression
 
 try:
-    if LooseVersion(sklearn.__version__) >= LooseVersion('0.18'):
-        from sklearn.model_selection import check_cv
-    # 0.18 > scikit-learn >= 0.16
-    else:
-        from sklearn.cross_validation import check_cv
+    #XXX this is going to be deprecated in sklearn 0.20
```

Comment:
The change is simple, not importing check_cv from model_selection, as it does not receive an X param and causes conflicts.
My bad, that comment was a personal message, I'll remove it.